### PR TITLE
Added lint for missing doc in metastore

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

**Comment by fulmicoton** [comment]
@mosuka  can you add the missing docs or make the guilty methods private or pub(crate) ?

**Comment by mosuka** [comment]
@fulmicoton 
Added missing docs in metastore.

File: quickwit-metastore/src/error.rs [hunk 1]
```diff
@@ -28,15 +28,25 @@ use thiserror::Error;
 /// Metastore error kinds.
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 pub enum MetastoreErrorKind {
+    /// The target split does not exist.
     DoesNotExist,
```
Comment (fulmicoton) [inline quickwit-metastore/src/error.rs:32]: Rename SplitDoesNotExist

File: quickwit-metastore/src/error.rs [hunk 2]
```diff
@@ -28,15 +28,25 @@ use thiserror::Error;
...
+    /// The target index exists.
     ExistingIndexUri,
+    /// The target split exists.
```
Comment (fulmicoton) [inline quickwit-metastore/src/error.rs:35]: The target split already exists.

File: quickwit-metastore/src/error.rs [hunk 3]
```diff
@@ -28,15 +28,25 @@ use thiserror::Error;
 /// Metastore error kinds.
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 pub enum MetastoreErrorKind {
+    /// The target split does not exist.
     DoesNotExist,
+    /// The target index exists.
```
Comment (fulmicoton) [inline quickwit-metastore/src/error.rs:33]: The target index already exists

File: quickwit-metastore/src/metastore.rs [hunk 4]
```diff
@@ -39,13 +39,16 @@ pub type IndexUri = String;
 /// A split ID.
 pub type SplitId = String;
 
+/// A file format version.
 pub static FILE_FORMAT_VERSION: &str = "0";
```
Comment (fulmicoton) [inline quickwit-metastore/src/metastore.rs:43]: Does it need to be public?

File: quickwit-metastore/src/metastore.rs [hunk 5]
```diff
@@ -39,13 +39,16 @@ pub type IndexUri = String;
...
 
+/// An index metadata carries all meta data about a split.
```
Comment (fulmicoton) [inline quickwit-metastore/src/metastore.rs:45]: i see no information about splits in this struct what am i missing?

File: quickwit-metastore/src/metastore/single_file_metastore.rs [hunk 6]
```diff
@@ -36,6 +36,7 @@ use crate::{
     SplitMetadata, SplitState, FILE_FORMAT_VERSION,
 };
 
+/// A metadata filename.
```
Comment (fulmicoton) [inline quickwit-metastore/src/metastore/single_file_metastore.rs:39]: The metadata filename

Comment (fulmicoton) [reply quickwit-metastore/src/metastore/single_file_metastore.rs:39]: this comment is useless.

**Review by fulmicoton — COMMENTED** [review]
Added some comments

File: quickwit-metastore/src/metastore.rs (hunk 5 above)
Comment (mosuka) [reply quickwit-metastore/src/metastore.rs:45]: Sorry, I made a mistake.

**Comment by mosuka** [comment]
@fulmicoton 
Thank you for reviewing. Please take another look.

**Pull request merged**